services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentsController.kt

```diff
@@ -0,0 +1,113 @@
+package fi.hel.haitaton.hanke.liitteet
+
+import fi.hel.haitaton.hanke.*
+import fi.hel.haitaton.hanke.permissions.PermissionCode
+import fi.hel.haitaton.hanke.permissions.PermissionService
+import java.util.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.multipart.MultipartFile
+
+@RestController
+@RequestMapping("/liitteet")
+class AttachmentsController(
+    @Autowired private val attachmentService: AttachmentService,
+    @Autowired private val hankeService: HankeService,
+    @Autowired private val permissionService: PermissionService,
+) {
+
+    @GetMapping("/hanke/{hankeTunnus}")
+    fun getAttachments(@PathVariable hankeTunnus: String): List<HankeAttachment> {
+        val currentUserId = currentUserId()
+        val hankeId =
+            hankeService.getHankeId(hankeTunnus) ?: throw HankeNotFoundException(hankeTunnus)
+
+        if (!permissionService.hasPermission(hankeId, currentUserId, PermissionCode.VIEW)) {
+            throw HankeNotFoundException(hankeTunnus)
+        }
+
+        val hankeAttachments = attachmentService.getHankeAttachments(hankeTunnus)
+        return hankeAttachments
+    }
+
+    @PostMapping("/hanke/{hankeTunnus}")
+    fun postAttachment(
+        @PathVariable hankeTunnus: String,
+        @RequestParam("liite") attachment: MultipartFile
+    ): HankeAttachment {
+        val currentUserId = currentUserId()
+        val hankeId =
+            hankeService.getHankeId(hankeTunnus) ?: throw HankeNotFoundException(hankeTunnus)
+
+        if (!permissionService.hasPermission(hankeId, currentUserId, PermissionCode.EDIT)) {
+            throw HankeNotFoundException(hankeTunnus)
+        }
+
+        val newAttachment = attachmentService.add(hankeTunnus, attachment)
+        return newAttachment
+    }
+
+    @GetMapping("/{uuid}")
+    fun getAttachment(@PathVariable uuid: UUID): HankeAttachment {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+
+        if (
+            info.hankeId == null ||
+                !permissionService.hasPermission(info.hankeId!!, currentUserId, PermissionCode.VIEW)
+        ) {
+            throw HankeNotFoundException("")
+        }
+
+        return info
+    }
+
+    @GetMapping("/{uuid}/data")
+    fun getAttachmentData(@PathVariable uuid: UUID): ByteArray {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+
+        if (
+            info.hankeId == null ||
+                !permissionService.hasPermission(info.hankeId!!, currentUserId, PermissionCode.VIEW)
+        ) {
+            throw HankeNotFoundException("")
+        }
+
+        return attachmentService.getData(uuid)
```

Comment:
Should the `Content-Disposition` header when sending the data? To tell what the name of this file is? Or will the frontend handle all of that?